### PR TITLE
FW-2795 Call API for CSAT Rating for Plan Restriction [Chat Widget]

### PR DESCRIPTION
**Pull request by Ojashwa**
### What do you want to achieve?
- Added restriction for CSAT according to plan subscription.

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [ ] I have compared it with mocks and all design elements are the same.
- [x] I have tested it in IE Browser.

### How was the code tested?

- Check for CSAT according to plan subscription.
```{
    "platform": "kommunicate",
    "message": "We are resolving this conversation. Please reach out to us in case of any more queries.",
    "metadata": {
        "actionRequest": "askCSAT"
    }
}
```
---
https://user-images.githubusercontent.com/22856546/134193101-818d6e18-038a-42c5-b008-006d9002f51e.mov

-
https://user-images.githubusercontent.com/22856546/134192816-095d0b4f-acf9-4fe0-81a9-a0b90354fdaa.mov

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/kommunicateCommons.js [hunk 1]
```diff
@@ -118,13 +118,21 @@ function KommunicateCommons() {
...
+        
+        var extractedUrl = message ? message.match(urlReg) : '';
         return message && extractedUrl
             ? this.isUrlValid(extractedUrl[0])
                 ? extractedUrl[0]
                 : false
             : false;
     };
+    _this.isCsatAvailable = function (appSettings) {
+        if (appSettings && appSettings.code == 'SUCCESS' && appSettings.response.isCsatAvailable) {
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicateCommons.js:130]: just this should be fine? 
return appSettings && appSettings.code == 'SUCCESS' && appSettings.response.isCsatAvailable;

File: webplugin/js/app/km-utils.js [hunk 2]
```diff
@@ -424,6 +424,23 @@ KommunicateUtils = {
             );
         }
     },
+    getAppSettings: function (appId,callback) {
+        var url =
+            KM_PLUGIN_SETTINGS.kommunicateApiUrl +
+            '/users/v2/chat/plugin/settings?appId=' +
```
Comment (devashishmamgain) [inline webplugin/js/app/km-utils.js:430]: is this a new settings API call? there must be already some code calling the settings API

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Review by Prochnost55 — CHANGES_REQUESTED** [review]
appSettings are already available when widget is initialized.

File: webplugin/js/app/km-utils.js (hunk 2 above)
Comment (Ojashwa) [reply webplugin/js/app/km-utils.js:430]: This call is to check if the user has CSAT in their plan from the backend.
what if the user overwrites appSettings after the installation of the script.
Let say he/she changes `isCsatAvailable ` in `kommunicateSettings ` inside installation script.

**Review by devashishmamgain — APPROVED** [review]

**Review by Prochnost55 — APPROVED** [review]
Mention In description to check with csat payload

**Review by devashishmamgain — APPROVED** [review]

**Review by ChannuBS — APPROVED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Review by ChannuBS — APPROVED** [review]

**Review by Prochnost55 — APPROVED** [review]

**Pull request merged**